zio-http/src/main/scala/zio/http/netty/socket/NettySocketProtocol.scala

```diff
@@ -27,47 +27,47 @@ import io.netty.handler.codec.http.websocketx.{
+      .dropPongFrames(webSocketConfig.dropPongFrames)
 
-  def serverBuilder(socketProtocol: SocketProtocol): WebSocketServerProtocolConfig.Builder =
+  def serverBuilder(webSocketConfig: WebSocketConfig): WebSocketServerProtocolConfig.Builder =
     WebSocketServerProtocolConfig
       .newBuilder()
       .checkStartsWith(true)
       .websocketPath("")
-      .subprotocols(socketProtocol.subprotocols.orNull)
-      .handshakeTimeoutMillis(socketProtocol.handshakeTimeoutMillis)
-      .forceCloseTimeoutMillis(socketProtocol.forceCloseTimeoutMillis)
-      .handleCloseFrames(socketProtocol.handleCloseFrames)
-      .sendCloseFrame(closeStatusToNetty(socketProtocol.sendCloseFrame))
-      .dropPongFrames(socketProtocol.dropPongFrames)
-      .decoderConfig(socketDecoderToNetty(socketProtocol.decoderConfig))
+      .subprotocols(webSocketConfig.subprotocols.orNull)
+      .handshakeTimeoutMillis(webSocketConfig.handshakeTimeoutMillis)
+      .forceCloseTimeoutMillis(webSocketConfig.forceCloseTimeoutMillis)
+      .handleCloseFrames(webSocketConfig.handleCloseFrames)
+      .sendCloseFrame(closeStatusToNetty(webSocketConfig.sendCloseFrame))
+      .dropPongFrames(webSocketConfig.dropPongFrames)
+      .decoderConfig(socketDecoderToNetty(webSocketConfig.decoderConfig))
 
-  private def closeStatusToNetty(closeStatus: CloseStatus): WebSocketCloseStatus =
+  private def closeStatusToNetty(closeStatus: WebSocketConfig.CloseStatus): WebSocketCloseStatus =
     closeStatus match {
-      case CloseStatus.NormalClosure         => WebSocketCloseStatus.NORMAL_CLOSURE
-      case CloseStatus.EndpointUnavailable   => WebSocketCloseStatus.ENDPOINT_UNAVAILABLE
-      case CloseStatus.ProtocolError         => WebSocketCloseStatus.PROTOCOL_ERROR
-      case CloseStatus.InvalidMessageType    => WebSocketCloseStatus.INVALID_MESSAGE_TYPE
-      case CloseStatus.InvalidPayloadData    => WebSocketCloseStatus.INVALID_PAYLOAD_DATA
-      case CloseStatus.PolicyViolation       => WebSocketCloseStatus.POLICY_VIOLATION
-      case CloseStatus.MessageTooBig         => WebSocketCloseStatus.MESSAGE_TOO_BIG
-      case CloseStatus.MandatoryExtension    => WebSocketCloseStatus.MANDATORY_EXTENSION
-      case CloseStatus.InternalServerError   => WebSocketCloseStatus.INTERNAL_SERVER_ERROR
-      case CloseStatus.ServiceRestart        => WebSocketCloseStatus.SERVICE_RESTART
-      case CloseStatus.TryAgainLater         => WebSocketCloseStatus.TRY_AGAIN_LATER
-      case CloseStatus.BadGateway            => WebSocketCloseStatus.BAD_GATEWAY
-      case CloseStatus.Empty                 => WebSocketCloseStatus.EMPTY
-      case CloseStatus.AbnormalClosure       => WebSocketCloseStatus.ABNORMAL_CLOSURE
-      case CloseStatus.TlsHandshakeFailed    => WebSocketCloseStatus.TLS_HANDSHAKE_FAILED
-      case CloseStatus.Custom(code, message) => new WebSocketCloseStatus(code, message)
+      case WebSocketConfig.CloseStatus.NormalClosure         => WebSocketCloseStatus.NORMAL_CLOSURE
```

Comment:
Does it make sense for a `Channel` to have a `def close(s: CloseStatus)` instead of this being part of config?

Reply:
It is part of the configuration in the underlying Netty implementation so I think this is going to be hard to do.